ibeaconscanner/src/main/java/mobi/inthepocket/android/beacons/ibeaconscanner/BluetoothScanBroadcastReceiver.java

```diff
@@ -0,0 +1,190 @@
+    /**
+     * Tag used to report that a beacon was no longer detected during a scan.
+     */
+    public static final String IBEACON_SCAN_BEACON_EXITED = "ibeacon_scan_beacon_detection_exit";
+
+    private static final String TAG = "BluetoothScanBrdcstRcvr";
+    private BeaconsSeenManager beaconsSeenManager;
+    private Context context;
+    private Class<?> targetService;
+    private FirebaseJobDispatcher dispatcher;
+    private long beaconExitTimeoutInMillis;
+
+    /**
+     * @see BroadcastReceiver#onReceive(Context, Intent)
+     */
+    @Override
+    public void onReceive(final Context context, final Intent intent)
+    {
+        this.dispatcher = new FirebaseJobDispatcher(new GooglePlayDriver(context));
+        this.beaconExitTimeoutInMillis = intent.getLongExtra(IBEACON_SCAN_EXITED_TIMEOUT_MS, BuildConfig.BEACON_EXIT_TIME_IN_MILLIS);
+        final BeaconsSeenProvider provider = new BeaconsSeenProvider(context);
+        this.beaconsSeenManager = new BeaconsSeenManager(provider, this.beaconExitTimeoutInMillis);
+        this.context = context;
+        final String className = intent.getStringExtra(IBEACON_SCAN_LAUNCH_SERVICE_CLASS_NAME);
+        try
+        {
+            this.targetService = TextUtils.isEmpty(className) ? null : Class.forName(className);
+        }
+        catch (final ClassNotFoundException e)
+        {
+            Log.e(TAG, "Target class was not found. Ensure you're passing the fully qualified namespace.", e);
+        }
+
+        final int bleCallbackType = intent.getIntExtra(this.getCallbackIntentKey(), -1);
+        if (bleCallbackType != -1)
+        {
+            final int errorCode = intent.getIntExtra(this.getErrorCodeIntentKey(), -1); // e.g.  ScanCallback.SCAN_FAILED_INTERNAL_ERROR
+            if (errorCode != -1)
+            {
+                Log.w(TAG, "Passive background scan failed.  Code; " + errorCode);
+            }
+            final Iterable<ScanResult> scanResults = intent.getParcelableArrayListExtra(this.getListScanResultIntentKey());
+            for (final ScanResult scanResult : scanResults)
+            {
+                this.processScanResult(scanResult);
+            }
+        }
+
+        provider.destroy();
+    }
+
+    private void processScanResult(final ScanResult scanResult)
+    {
+        Log.d(TAG, "In the range of beacon: " + scanResult.toString());
+
+        // get Scan Record byte array (Be warned, this can be null)
+        if (scanResult.getScanRecord() != null)
+        {
+            final Pair<Boolean, Integer> isBeacon = BeaconUtils.isBeaconPattern(scanResult);
+
+            if (isBeacon.first)
+            {
+                final Beacon beacon = BeaconUtils.createBeaconFromScanRecord(scanResult.getScanRecord().getBytes(), isBeacon.second);
+
+                // see if the beacon was not yet triggered
+                if (!this.beaconsSeenManager.hasBeaconBeenTriggered(beacon))
```

Comment:
maybe I'm overlooking something, but if we receive a trigger for a beacon we've already seen, don't we have to reset the exit timer? or does this happen in another place?

Reply:
Did not look into this. @nielsdelporte

Reply:
You're right, @qbraet . The dispatcher should update the existing job if that is te case.
This use case should be implemented here, but has not been added.